pilot/pkg/model/context.go

```diff
@@ -308,12 +309,33 @@ func (s *StringBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// ProxyConfig can only be marshalled using (gogo) jsonpb. However, the rest of node meta is not a proto
+// To allow marshalling, we need to define a custom type that calls out to the gogo marshaller
+type JsonProxyConfig meshconfig.ProxyConfig
+
+func (s JsonProxyConfig) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	pc := meshconfig.ProxyConfig(s)
+	if err := (&gogojsonpb.Marshaler{}).Marshal(&buf, &pc); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (s *JsonProxyConfig) UnmarshalJSON(data []byte) error {
+	pc := (*meshconfig.ProxyConfig)(s)
+	return gogojsonpb.Unmarshal(bytes.NewReader(data), pc)
+}
+
 // NodeMetadata defines the metadata associated with a proxy
 // Fields should not be assumed to exist on the proxy, especially newly added fields which will not exist
 // on older versions.
 // The JSON field names should never change, as they are needed for backward compatibility with older proxies
 // nolint: maligned
 type NodeMetadata struct {
+	// ProxyConfig defines the proxy config specified for a proxy.
+	ProxyConfig *JsonProxyConfig `json:"PROXY_CONFIG,omitempty"`
```

Comment:
Since pilot also reads MeshConfig, hence reading "global" proxy config too, where and what should be used in the control plane?